Automattic/jetpack
# Jetpack mu and wpcomsh - hide wpcom items for non wpcom site users, hide hosting card from non-admins

Fixes # Automattic/dotcom-forge#7892

## Proposed changes:


Hides various wpcom items in wp-admin when the current user is not connected to wpcom:

* Hosting card on the dashboard. Note - this PR also hides this for connected users who are not admins
<img width="634" alt="Screenshot 2024-07-02 at 2 54 33 PM" src="https://github.com/Automattic/jetpack/assets/28742426/f2c72012-444b-4c18-89e5-245db36ed777">

* WPcom command palette
<img width="696" alt="Screenshot 2024-07-02 at 2 54 56 PM" src="https://github.com/Automattic/jetpack/assets/28742426/f5bee792-8e85-4e94-8b56-4f68828e5196">


* Themes banner (seen at themes -> add new)
<img width="1284" alt="Screenshot 2024-07-02 at 2 55 19 PM" src="https://github.com/Automattic/jetpack/assets/28742426/808fc590-3aef-479d-811d-71d584c7e4e9">


* themes showcase sidebar link.
<img width="161" alt="Screenshot 2024-07-02 at 2 55 31 PM" src="https://github.com/Automattic/jetpack/assets/28742426/8d747986-d23a-4610-9c94-4a82270f812c">

* plugins marketplace sidebar link
<img width="163" alt="Screenshot 2024-07-02 at 2 55 47 PM" src="https://github.com/Automattic/jetpack/assets/28742426/8f7e1ce3-f186-4d61-a04d-b1d94e0d049c">


* plugins banner under 'add plugins'
<img width="1289" alt="Screenshot 2024-07-02 at 2 55 57 PM" src="https://github.com/Automattic/jetpack/assets/28742426/917a53f2-50f1-414d-bfe5-ff63084b24be">


### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Set up a woa dev blog.
* install the jetpack beta tester plugin and ensure your test site has the JETPACK_MU_WPCOM_LOAD_VIA_BETA_PLGUIN constant defined (both noted in github-actions comment below).
* In the jetpack beta tester plugin's settings, find "jetpack beta tester" and activate the "bleeding edge" version. This is necessary to activate this branch on wpcomsh here.
* Next, activate this branch in both "WordPress.com Features" and "WordPress.com Site Helper" sections of the jetpack beta tester plugin.
* Verify all of the above items shown in the description still show up as expected in wp-admin.
* Visit Users-> add new user. Create a new user and password for the site, do not invite them to wpcom.
* Open wp-admin for this site in an incognito tab. Instead of logging in via wordpress.com, login via the new site user/pw combo you created.
* In wp-admin, verify the above items are not visible.

* Add another user who is connected to wpcom, and add them as something that is not an administrator (i did editor).
* In the wp-admin dashboard, verify the site hosting management card is not available to the non-admin user.
